File: app/src/main/java/com/MohamedTaha/Imagine/New/service/IntentModule.java
package com.MohamedTaha.Imagine.New.service;

import android.content.Intent;

import com.MohamedTaha.Imagine.New.scope.ScopeActivity;
import com.MohamedTaha.Imagine.New.scope.ScopeNavigationDrawaberActivity;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class IntentModule {
    //@ScopeActivity
    @Provides
    public Intent getIntent(){
        return new Intent();
    }
}

File: app/src/main/java/com/mohamedtaha/imagine/hilt/module/IntentModule.kt
package com.mohamedtaha.imagine.hilt.module

import android.content.Intent
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import javax.inject.Singleton

@Module
@InstallIn(SingletonComponent::class)
class IntentModule {
    @Provides
    fun intentProvide():Intent = Intent()
}